src/lib.rs

```diff
@@ -141,11 +194,25 @@ impl<'a> Iterator for SMRecIterator<'a> {
         //     uint16: NumLocations
         let num_locs = itry!(cursor.read_u16::<NativeEndian>());
         //     Location[NumLocations] { ... }
+        let loc_iter = SMLocIterator {
+            elf_file: self.elf_file,
+            cursor: None,
+            start_pos: cursor.position(),
+            num_locs: num_locs
+        };
+
+        let mut locs = Vec::with_capacity(num_locs as usize);
+        for loc in loc_iter {
+            locs.push(loc.expect("malformed location"))
+        }
+
         let skip_locs_sz = u64::from(u32::from(num_locs) * u32::from(SIZE_LOC_ENTRY));
         itry!(cursor_skip(&mut cursor, skip_locs_sz as i64));
+
         //     uint32: Padding (only if required to align to 8 byte)
         //     uint16: Padding
         itry!(cursor_align8(&mut cursor));
+        itry!(cursor_skip(&mut cursor, 2));
```

Comment:
Oops, so I missed skipping the `uint16` pad?

Reply:
Yeah ;)